#5: Spack for Teuchos

marcinwrobel1986 opened the pull request:
### THIS PR:
- spack package for Teuchos

Commit:
#5: - spack package for Teuchos

amd-toolchain-support committed:
Add version 3.2 of AMD libraries (#31538)

Commit:
#5: - added TPL support

Commit:
#5: - back to proper version

mperrinel commented:
@marcinwrobel1986 Seems good. 3 things for me to verify/change:
1) The QT tpl is named QT and not Qt.
2) The teuchos **test** should be enabled as a variant of spack and not be enabled automatically
3) The GIT version is missing, You added the URL but removed the GIT. We should used GIT as a preferred version and specify a "CI" version to use the URL (which you could use on the CI).

stephenmsachs committed:
[py-numpy] Newer versions do not build with `icc` (#31637)

* [py-numpy] Newer versions do not build with `icc`

Workaround until https://github.com/numpy/numpy/issues/22011 is solved.

* [py-numpy] Remove whitespaces

nicholas-sly committed:
py-torch: use immutable URLs for open-ce patches (#31672)

* Update open-ce patches for py-torch to us immutable URLs.  Update magma dependency specs to be more explicit.

* Address comments for PR regarding URLs and conflicting variants.

Co-authored-by: Nicholas Cameron Sly <sly1@llnl.gov>

eschnett committed:
mpitrampoline: New version 4.2.0 (#31683)

alalazo committed:
Use pkg_cls in spack.mirror.get_all_versions (#31636)

fixes #31627

spack.mirror.get_all_versions now uses the package class
instead of the package object in its implementation.

Ensure spec is concrete before staging for mirrors

haampie committed:
autotools: add -I flag when non-standard libtool (#31677)

When
1. Spack installs libtool,
2. system libtool is installed too, and
3. system automake is used

Spack passes system automake's `-I <prefix>` flag to itself, even though
it's a default search path. This takes precedence over spack's libtool
prefix dir. This causes the wrong `libtool.m4` file to be used (since
system libtool is in the same prefix as system automake).

And that leads to error messages about incompatible libtool, something
something LT_INIT.

hseara committed:
gromacs: Added version 2021.6 (#31692)

haampie committed:
Mark external as explicit only when installing explicitly (#31665)

haampie committed:
coreutils: gnulib patch (#31693)

Co-authored-by: snehring <snehring@iastate.edu>

haampie committed:
llvm: fix patch already applied on main branch (#31694)

haampie committed:
clingo patch fixed in patch release (#31690)

netromdk committed:
py-vermin: add latest version 1.4.1 (#31708)

IshaanDesai committed:
[py-pyprecice] add v2.4.0.0 (#31697)

* Add py-pyprecice v2.4.0.0

* Adding lower bound to pip dependency

* Removing when=@:2.1 from dependencies packaging and pip

adamjstewart committed:
Sphinx 5.1.0 breaks doc builds (#31709)

Commit:
#5: - trying add program_options for boost

adamjstewart committed:
py-sphinx: add v5.1.0 (#31711)

adamjstewart committed:
py-geopandas: add v0.11.1 (#31710)

adamjstewart committed:
py-torchmetrics: add v0.9.3 (#31707)

adamjstewart committed:
py-torchdata: add new package (#31704)

* py-torchdata: add new package

* third_party deps only in 0.4+

snehring committed:
wget: replacing prior workaround with proper fix (#31699)

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

haampie committed:
Fix stage with uri params (#31618)

* don't determine extension of a url

* parse url to drop query params from filename

haralmha committed:
llvm: Add missing include for version range 8 to 11 (#31639)

Paul-Ferrell committed:
LLVM requires swig version > 3 (#31625)

psakievich committed:
GCC: Add bootstrap conflict for M1 (#31189)

snehring committed:
gdal: fixing typo for libxml2 (#31734)

rbberger committed:
new package: singularity-eos (#31500)

eugeneswalker committed:
openblas %oneapi: patch f_check for proper linkage (#31737)

stephenmsachs committed:
Fix `self.extension` for symlinks (#31743)

`self.archive_file` is (among others) a symlink to a tarball. `extension()` on a
symlink will result in no extension. This patch fixes the behavior introduced in
https://github.com/spack/spack/pull/31618.

Co-authored-by: Stephen Sachs <stesachs@amazon.com>

Commit:
llvm: Variant and dependency cleanup (#31331)

* llvm: Use variant when clauses for many of the expressed conflicts

* llvm: Remove the shared variant as it wasn't really used

* llvm: Remove unnecessary deps and make explicit the ones that are

* llvm: Cleanup patch conditions

* pocl: Update for llvm cleanup

* unit-test: update unparse package hash with the updated llvm package

* llvm: Fix ppc long double patching and add clarifying comments

haampie committed:
cmake: missing patch releases (#31676)

haampie committed:
containerize: fix concretization -> concretizer (#31594)

* containerize: fix concretization -> concretizer

* fix test

haampie committed:
environment.py: reduce # of locks further (#31643)

* environment.py: reduce # of locks further

haampie committed:
gnupg: bump and deprecate over cve (#31730)

haampie committed:
libgpg-error: bump (#31729)

haampie committed:
libgcrypt: bump (#31728)

haampie committed:
munge: 0.5.15 (#31727)

haampie committed:
json-c: 0.16 (#31726)

haampie committed:
jansson: new version, disable docs (#31724)

We don't have the build deps for the docs, so disable for now.

haampie committed:
libjwt: new versions, missing dep, disable examples (#31723)

haampie committed:
openmpi: +pmi implies schedulers=slurm (#31722)

eugeneswalker committed:
mpich@4.0.2 %oneapi: patch config.rpath (#31740)

eugeneswalker committed:
e4s ci stack: add spec: hdf5-vol-async (#31747)

* e4s ci stack: add spec: hdf5-vol-async

* hdf5-vol-async: add e4s tag

eugeneswalker committed:
cabana: add cajita variant (#31753)

* cabana: add cajita variant

* Update var/spack/repos/builtin/packages/cabana/package.py

Co-authored-by: Sam Reeve <6740307+streeve@users.noreply.github.com>

* cabana: cannot build +cajita ~mpi

Co-authored-by: Sam Reeve <6740307+streeve@users.noreply.github.com>

haampie committed:
curl: 7.84.0 (#31725)

melven committed:
itensor: new version, restrict hdf5 to :1.10 (#31764)

melven committed:
valgrind: new version (#31762)

melven committed:
tixi: new versions (#31761)

adamjstewart committed:
py-sphinx: add v5.1.1 (#31756)

jrmadsen committed:
Update omnitrace package with v1.3.0 (#31754)

wspear committed:
Tau must get GCC path from environment on Cray (#31751)

* Tau must get GCC path from environment on Cray

self.compiler doesn't provide the path to the gcc compiler when using cray cc and the spack internal compiler overrides the location in PATH. If possible get the location from the GCC_PATH variable instead.

* Fix flake8 issues

* Update package.py

duperray committed:
py-scipy: -fPIC appending trouble with gcc@10: (#31698)

`-fPIC` mandatory option squeezed by error, the
NPY_DISTUTILS_APPEND_FLAGS appears handy

Co-authored-by: duperrayp <duperrayp@u-un00s16835>

vvolkl committed:
herwig3, thepeg: correct lhapdfsets dependency, specify boost libs (#31744)

* herwig3: change lhapdfsets dependency type to build

These data sets are needed for a check during build, but due to the difficulty of versioning the datasets it is preferred not to keep the them in the run environment.

* herwig3: explicitly state needed boost libs

* thepeg: explicitly state needed boost libs

* style

* stylestyle

vvolkl committed:
dd4hep: add v1.21 (#31705)

kwryankrattiger committed:
WarpX: Add sensei insitu variant (#31542)

brugger1 committed:
VisIt: Update to VisIt 3.3.0. (#31654)

Andrew-Dunning-NNL committed:
ZSH: add lmod and local etc variants (#29651)

* zsh: add lmod and local etc support

* zsh: fix indentation

tmadlener committed:
lcio, podio, edm4hep: Add latest versions (#31749)

* lcio: add latest version

* edm4hep: Add latest versions

* podio: Add latest versions

daboehme committed:
Add Caliper v2.8.0 and rocm support (#31632)

* Add Caliper v2.8.0

* Add conflict for +rocm+cuda

streeve committed:
New ExaCA package (#31777)

eschnett committed:
c-blosc2: New version 2.2.0 (#31577)

eugeneswalker committed:
e4s ci: add slate +rocm (#31602)

amd-toolchain-support committed:
Adding Flamemaster product (#31126)

* Adding Flamemaster product

About FlameMaster: Flamemaster is an open source C++ program package for 0D
combustion and 1D laminar flame calculations.

* Addressed review comments and style tests errors

* Addressed style tests errors

* Incorporated spack style of coding

* Fixed following spack style error

[E228] missing whitespace around modulo operator

* Update package.py

addressing style issues

* Update package.py

* Changes to maintainers entry

mzuzek commented on var/spack/repos/builtin/packages/teuchos/package.py:
```diff
@@ -19,98 +19,76 @@
     maintainers = ['keitat']
 
-    version('1-teuchos-cmake', branch='1-teuchos-cmake', preferred=True)
+    version('0.1.0', sha256=@@@@@, preferred=True)
```
@marcinwrobel1986 Python can't parse this...

marcinwrobel1986 replied:
@mzuzek yes, it is substituted with calculated sha256. I will try to use `file://` and maybe try to remove sha256. When still no success, then I will at least put @@@@@ inside quotes like '@@@@@', so then it will not raised an error.

marcinwrobel1986 committed:
Merge branch 'spack:develop' into 5-spack-for-Teuchos-MW

eschnett committed:
mpitrampoline: New version 5.0.1 (#31772)

stephenmsachs committed:
Fix parsing of clean_url (#31783)

In #31618 the idea was to determine the file extension heuristically by dropping query params etc from a url and then consider it as a file path. That broke for URLs that only have query params like http://example.com/?patch=x as it would result in empty string as basename. This PR reverts to the old behavior of saving files as ?patch=x in that case.

Co-authored-by: Stephen Sachs <stesachs@amazon.com>

fspiga committed:
Adding NVIDIA HPC SDK 22.7 (#31770)

haampie committed:
gcc: fix unstable patch urls (#31784)

the patch urls dynamically generate a diff, which includes metadata
about the git version used, meaning they are not content-addressable.

instead ship the patches with spack.

brenthuisman committed:
New version of Arbor package (#31660)

manuelakuhn committed:
py-rsatoolbox: add 0.0.4 (#31785)

streeve committed:
Cabana: add test variants (#31776)

* Simplify Cabana build flags
* Add unit tests, examples, and performance options

haampie committed:
environment.py: only install root specs explicitly (#31645)

snehring committed:
sentieon-genomics: update version to 202112.05 (#31788)

q-posev committed:
New package: trexio (#31769)

* [WIP] first working Spack spec for TREXIO

* add patch to prepend -lhdf5_hl to LIBS

* Fix spack stype issues

* trexio: Fix hashes and hdf5 variant build

* trexio: Remove import from Spack

* trexio: Fix the isort error

* Apply suggestions from code review

By @tldahlgren

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>

glennpj committed:
new package: r-signac (#31703)

Also includes new package depdendency for r-rcpproll.

hainest committed:
Dyninst: add version 12.2.0 (#31789)

scottwittenburg committed:
Deprecate visit 3.3.0 because it is broken (#31790)

eugeneswalker committed:
e4s ci: add oneapi stack (#31781)

* e4s ci: add oneapi stack

* shorten padded_length to 256

* comment out pdt and add failure note

marcinwrobel1986 committed:
Merge branch 'spack:develop' into 5-spack-for-Teuchos-MW

marcinwrobel1986 commented:
Moved to another repo, closing
